src/validate_pyproject/plugins/__init__.py

```diff
@@ -20,12 +20,9 @@
     from importlib_metadata import EntryPoint, entry_points
 
 if typing.TYPE_CHECKING:
-    from ..types import Plugin, Schema
+    from typing import Protocol
```

Comment:
There's no savings to putting this in `typing.TYPE_CHECKING`, as `typing` is already imported. So I'd just move it up and then you can skip the `Protocol = object` hack.